sockeye/inference.py

```diff
@@ -1019,9 +1024,14 @@ def _get_inference_input(self,
         max_output_lengths = []  # type: List[int]
         for j, trans_input in enumerate(trans_inputs):
             num_tokens = len(trans_input)  # includes eos
-            max_output_lengths.append(self._get_max_output_length(num_tokens))
-            source_np[j, :num_tokens, 0] = tokens2ids(itertools.chain(trans_input.get_source_prefix_tokens(),
-                                                                      trans_input.tokens), self.source_vocabs[0])
+            primary_source_ids = tokens2ids(itertools.chain(trans_input.get_source_prefix_tokens(),
+                                                            trans_input.tokens), self.source_vocabs[0])
+            source_np[j, :num_tokens, 0] = primary_source_ids
+            length_np[j, 0] = num_tokens
+            if self.eop_id != C.INVALID_ID:
```

Comment:
I was trying to avoid the look-up operation in favor of latency.

Reply:
That's a good point. Would it make sense to make this check part of the get_prepended_token_length function. Its a function called lots of times (also in data preprocessing) and maybe its worth doing the check every single time to avoid the look up in the list.

Reply:
Sure, added.